core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -680,22 +697,23 @@ class Partition(val topicPartition: TopicPartition,
    * produce request.
    */
   def checkEnoughReplicasReachOffset(requiredOffset: Long): (Boolean, Errors) = {
-    leaderReplicaIfLocal match {
-      case Some(leaderReplica) =>
+    leaderLogIfLocal match {
+      case Some(leaderLog) =>
         // keep the current immutable replica list reference
         val curInSyncReplicas = inSyncReplicas
 
         if (isTraceEnabled) {
           def logEndOffsetString(r: Replica) = s"broker ${r.brokerId}: ${r.logEndOffset}"
-          val (ackedReplicas, awaitingReplicas) = curInSyncReplicas.partition { replica =>
+          val curInSyncReplicaObjects = (curInSyncReplicas - localBrokerId).map(getReplicaOrException)
```

Comment:
As the log was related to acks from replica, my thinking was that leader information in the log message doesn't matter. Now if this method can be invoked for non-leader case, then maybe this is useful (but then inSyncReplicas is empty set).

Or if we don't want to change behavior (or I am misreading code), I can rearrange code to include local logs information too. Let me know.

Reply:
I get what you're saying, but the leader conventionally counts as one ack. I think it's probably best to leave it as is to avoid any potential confusion.

Reply:
Done.